c/loops/array-1.yml

```diff
@@ -11,3 +11,8 @@ properties:
   - property_file: ../properties/coverage-branches.prp
   - property_file: ../properties/coverage-conditions.prp
   - property_file: ../properties/coverage-statements.prp
+
+options:
+  language: C
+  data_model: ILP32
```

Comment:
Is the decision final to add the intended data-model on the task-definition level, instead of the property-level?

I would still prefer

```yml
properties:
    - property_file: ../properties/termination.prp
      data_model: ILP32
      expected_verdict: true
# ... snip ...

options:
   language: C
```

Advantage: single task can have different properties and verdicts for different data models.  
Disadvantage: I guess we still need the `language: C` on the task-definition level?

I wanted to make these considerations explicit, but I'm fine with either because we didn't have programs with properties for both 32 and 64bit, so far.

Reply:
> Is the decision final to add the intended data model on the task-definition level, instead of the property-level?

Yes. Several years ago in the beginning of the discussion about task-definition files a more flexible prototype was made, and the experience with this prototype was that it is not worth the complexity to allow defining different variants of the same task in the same task-definition file (e.g., same program files, same property, but different compiler flags). For example, consider that if we do what you suggest, the tuple (task-definition file, property file) does no longer uniquely identify a task, we would need a new way of identifying tasks for example in tables.

Furthermore, having the option on task-definition level removes redundancy and is less error-prone in the common case where the values are the same for all properties, and is no less general, because one can always use several task-definition files if a property-specific value is necessary. On the other hand, having the `options` key per property is less general because then it is not possible to use it without property files, and allowing it in both places is also confusing (how should contradicting values be handled)?

Adding additional values like `data_model` directly to the property entry is not possible anyway because the latter is interpreted by `benchexec`, and the data that is not interpreted by it and only passed to the tool-info module must be clearly separated to avoid conflicts.

Reply:
Makes sense, thanks for the detailed explanation!